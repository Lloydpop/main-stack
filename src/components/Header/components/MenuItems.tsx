import homeIcon from "@/assets/images/homeIcon.png";
import chartIcon from "@/assets/images/chartIcon.png";
import moneyIcon from "@/assets/images/moneyIcon.png";
import groupIcon from "@/assets/images/groupIcon.png";
import widgetIcon from "@/assets/images/widgetIcon.png";
import notificationIcon from "@/assets/images/notificationIcon.png";
import chatIcon from "@/assets/images/chatIcon.png";
export const headerMenu = {
  menu_one: [
    { label: "Home", icon: homeIcon, variant: "transparent" },
    { label: "Analytics", icon: chartIcon, variant: "transparent" },
    { label: "Revenue", icon: moneyIcon, variant: "primary" },
    { label: "CRM", icon: groupIcon, variant: "transparent" },
    { label: "Apps", icon: widgetIcon, variant: "transparent" },
  ],
  menu_two: [{ icon: notificationIcon }, { icon: chatIcon }],
};
