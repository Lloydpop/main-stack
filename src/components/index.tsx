export { Wrapper } from "./Wrapper/Wrapper";
export { Button } from "./Button/Button";
export { Avatar } from "./Avatar/Avatar";
export { Heading } from "./Heading/Heading";
export { Badge } from "./Badge/Badge";
export { BasicDateRangePicker } from "./DatePicker/DatePicker";
export { Dropdown } from "./Dropdown/Dropdown";
export { EmptyState } from "./EmptyState/EmptyState";
