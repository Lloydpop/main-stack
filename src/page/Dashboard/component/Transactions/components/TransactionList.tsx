import { Badge, Heading } from "../../../../../components";
import arrowUp from "../../../../../assets/images/arrowupIcon.png";
import arrowDown from "../../../../../assets/images/arrowdownIcon.png";
import { formatAsUSD, formatDateToTextFormat } from "../../../../../utils";
export interface TransactionListProps {
  data: {
    metadata: { product_name: string; name: string };
    status: string;
    amount: string;
    date: string;
  }[];
}

export const TransactionList = ({ data }: TransactionListProps) => {
  return (
    <div className="space-y-6 border-t border-[#80808023] py-6">
      {data?.map((transaction, i) => (
        <div key={i} className="flex lg:flex-row flex-col justify-between">
          <div className="flex items-center gap-4 ">
            <div
              className={`w-[48px] h-[48px] rounded-full flex items-center justify-center ${
                transaction?.status === "successful"
                  ? "bg-[#E3FCF2]"
                  : "bg-[#F9E3E0]"
              }`}
            >
              <img
                src={transaction?.status === "successful" ? arrowDown : arrowUp}
                className="w-4 h-4 object-contain"
                alt="arrow icon"
              />
            </div>
            <div>
              <p className="font-semibold">
                {transaction?.metadata?.product_name ?? "Annonymous Product"}
              </p>
              <p className="text-[#56616B] mt-1">
                {transaction?.metadata?.name ?? "Annonymous"}
              </p>
              <Badge status={transaction?.status}>{transaction?.status}</Badge>
            </div>
          </div>
          <div className="lg:ml-0 lg:mt-0 ml-16 mt-6">
            <Heading size="16">{formatAsUSD(transaction?.amount)}</Heading>
            <p className="mt-1">{formatDateToTextFormat(transaction?.date)}</p>
          </div>
        </div>
      ))}
    </div>
  );
};
