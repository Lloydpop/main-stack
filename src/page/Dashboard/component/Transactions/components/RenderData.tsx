import { EmptyState } from "../../../../../components";
import { TransactionList } from "./TransactionList";
interface RenderDataProps {
  data: {
    metadata: { product_name: string; name: string };
    status: string;
    amount: string;
    date: string;
  }[];
  action: () => void;
}
export const RenderData = ({ data, action }: RenderDataProps) => {
  return (
    <div>
      {data?.length === 0 ? (
        <EmptyState
          action={{
            label: "Clear filters",
            action: () => {
              action();
            },
          }}
          description="Change your filters to see more results, or add a new product."
          title="No matching transaction found for the selected filter"
        />
      ) : (
        <TransactionList data={data} />
      )}
    </div>
  );
};
