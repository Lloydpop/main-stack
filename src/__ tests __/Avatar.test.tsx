import { render } from "@testing-library/react";
import { Avatar } from "../components";

describe("Avatar component", () => {
  test("renders Avatar component correctly", () => {
    const name = "John Doe";
    const { container } = render(<Avatar name={name} />);
    expect(container).toMatchSnapshot();
  });
});
