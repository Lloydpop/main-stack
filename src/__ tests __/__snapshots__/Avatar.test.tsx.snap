// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders Avatar component correctly 1`] = `
<div>
  <div
    class="bg-gradient-to-r from-gray-700 via-gray-600 to-gray-900  w-[32px] h-[32px] rounded-full text-white uppercase flex justify-center items-center"
  >
    <span
      class="initials"
    >
      JD
    </span>
  </div>
</div>
`;
